packages/core/src/worker/helpers/exponentialBackOffRetryStrategy.ts

```diff
@@ -0,0 +1,26 @@
+import { Observable, timer } from 'rxjs';
+import { retry } from 'rxjs/operators';
+import type { Page, SourceSchedule } from '@jpmorganchase/mosaic-types';
+
+export const exponentialBackOffRetryStrategy =
+  (schedule: SourceSchedule) => (attempts: Observable<Page[]>) =>
+    attempts.pipe(
+      retry({
+        count: schedule.retryEnabled ? schedule.maxRetries : 0,
+        delay: (_error, retryIndex) => {
+          const retryDelayMs = schedule.retryDelayMins * 60000;
+          // eslint-disable-next-line prefer-exponentiation-operator, no-restricted-properties
+          const backOffTime = Math.pow(2, retryIndex - 1) * retryDelayMs;
+          const remainingAttempts = schedule.maxRetries - retryIndex;
+          console.warn('[Mosaic] Retrying failed source');
```

Comment:
does this need to identify the source

Reply:
Can't here. The id is a generated symbol and can't be passed to the source worker thread via worker data as it's incompatible with structured cloning.

Reply:
We actually have a `name` property that isn't a symbol being passed to the worker.  Pushed a commit that uses that:

Example log:

```
[Mosaic] Source 'site#91fb4fd4' started in 1386ms - awaiting first docs snapshot
[Mosaic] Source 'site#91fb4fd4' received first docs snapshot 0.001s after starting.
[Mosaic] Retrying failed source: site#91fb4fd4
[Mosaic] Attempt (1/2). 1 attempt remaining
[Mosaic] Retry delay: 1 mins for source: site#91fb4fd4
```